tiledb/sm/query/readers/result_tile.h

```diff
@@ -431,7 +441,8 @@ class ResultTile {
       const std::string& name,
       const TileSizes tile_sizes,
       const TileData tile_data,
-      unsigned dim_idx);
+      unsigned dim_idx,
+      shared_ptr<MemoryTracker> memory_tracker);
```

Comment:
ResultTile doesn't have a `memory_tracker_` variable. Just to make sure I didn't give up too easily when I wrote this PR I just went back and tried adding one. After a couple hours of work I've got a 300 line diff that doesn't compile because I can't figure out how subclass constructor deferral on move construction is supposed to work.

Reply:
Would it help if I make a branch that disables all move constructors for Tile, ResultTile, etc?

Reply:
Sure would it turns out. That made the changes to ResultTile quite a bit easier to handle. This PR is updated with the ResultTile instrumentation.